apps/entry/stats.py

```diff
@@ -0,0 +1,264 @@
+from analysis_framework.models import Widget, Filter
+
+from .models import Entry, Attribute
+
+SUPPORTED_WIDGETS = [
+    'matrix1dWidget', 'matrix2dWidget', 'scaleWidget', 'multiselectWidget', 'organigramWidget', 'geoWidget',
+    'conditionalWidget',
+]
+
+
+def _get_widget_info(config, widgets, widget_name, skip_data=False):
+    widget = widgets[config[widget_name]['pk']]
+
+    def _return(data):
+        return {
+            '_widget': widget,
+            'pk': widget.pk,
+            'config': config[widget_name],
+            'data': data,
+        }
+
+    if skip_data:
+        return _return(None)
+
+    if widget.widget_id == 'organigramWidget':
+        w_filter = Filter.objects.filter(
+            widget_key=widget.key,
+            analysis_framework_id=widget.analysis_framework_id,
+        ).first()
+        return _return(w_filter.properties if w_filter else None)
+
+    data = widget.properties['data']
+    # TODO: Add validator for this
+    selectors = config[widget_name].get('selectors')
+    if selectors:
+        for selector in selectors + ['properties', 'data']:
+            data = data[selector]
+    return _return(data)
+
+
+def _get_attribute_widget_value(cd_widget_map, w_value, widget_type, widget_pk=None):
+    if widget_type in ['scaleWidget', 'multiselectWidget', 'organigramWidget', 'geoWidget']:
+        return w_value
+    elif widget_type == 'conditionalWidget':
+        cd_config = cd_widget_map.get(widget_pk)
+        if cd_config is None:
+            return
+        selected_widget_key = cd_config['widget_key']
+        selected_widget_type = cd_config['widget_type']
+        selected_widget_pk = cd_config.get('widget_pk')
+        return _get_attribute_widget_value(
+            cd_widget_map,
+            w_value[selected_widget_key]['data']['value'],
+            selected_widget_type,
+            selected_widget_pk,
+        ) if w_value.get(selected_widget_key) else None
+    elif widget_type in ['matrix1dWidget', 'matrix2dWidget']:
+        context_keys = [
+            f'{widget_pk}-{_value}'
+            for _value in (
+                w_value.keys() if isinstance(w_value, dict) else []
+            )
+        ]
+        sectors_keys = []
+        if widget_type == 'matrix2dWidget':  # Collect sector data from here
+            sectors_keys = [
+                [f'{widget_pk}-{pillar_key}', subpillar_key, sector_key]
+                for pillar_key, pillar in w_value.items()
+                for subpillar_key, subpillar in pillar.items()
+                for sector_key in subpillar.keys()
+            ]
+        return {
+            'context_keys': context_keys,
+            'sectors_keys': sectors_keys,
+        }
+
+
+def _get_attribute_data(collector, attribute, cd_widget_map):
+    widget_type = attribute.widget.widget_id
+    widget_pk = attribute.widget.pk
+    data = attribute.data
+
+    if widget_type not in SUPPORTED_WIDGETS or data is None or data.get('value') is [None, {}, []]:
```

Comment:
Maybe it should be `is in [None, {}, []]` instead of `is [None, {}, []]` ?

Reply:
Done